dev/local/Makefile

```diff
@@ -0,0 +1,136 @@
+#   test-subdir=  Only runs tests in the directory provided here, e.g.
+#                 repos/delphi/delphi-epidata/tests/acquisition/covidcast
+
+
+# Set optional argument defaults
+ifdef pdb
+	override pdb=--pdb
+else
+	pdb=
+endif
+
+ifdef test-subdir
+else
+	test-subdir=repos/delphi/delphi-epidata/tests repos/delphi/delphi-epidata/integrations
+endif
+
+# Set your shell here
+SHELL:=/bin/sh
+
+# Get the Makefile's absolute path: https://stackoverflow.com/a/324782/4784655
+# (if called from symlink, the path is the location of the symlink)
+NOW:=$(shell date "+%Y-%m-%d")
+CWD:=$(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+LOG_WEB:=$(CWD)docker-logs/delphi_web_epidata_$(NOW).log
+LOG_DB:=$(CWD)docker-logs/delphi_database_epidata_$(NOW).log
+
+
+.PHONY=web
+web:
+	@# Stop container if running
+	@-docker ps | grep delphi_web_epidata && docker stop delphi_web_epidata
+
+	@# Setup virtual network if it doesn't exist
+	@-docker network ls | grep delphi-net || docker network create --driver bridge delphi-net
+
+	@# Build the web_epidata image
+	@cd repos/delphi/delphi-epidata;\
+	docker build -t delphi_web_epidata -f ./devops/Dockerfile .;\
+	cd ../../../
+
+	@# Run the web server
+	@docker run --rm -p 127.0.0.1:10080:80 \
+		--env "SQLALCHEMY_DATABASE_URI=mysql+mysqldb://user:pass@delphi_database_epidata:3306/epidata" \
+		--env "FLASK_SECRET=abc" --env "FLASK_PREFIX=/epidata" \
+		--network delphi-net --name delphi_web_epidata \
+		delphi_web_epidata >$(LOG_WEB) 2>&1 &
+
+.PHONY=db
+db:
+	@# Stop container if running
+	@-docker ps | grep delphi_database_epidata && docker stop delphi_database_epidata
+
+	@# Only build prereqs if we need them
+	@-docker images delphi_database | grep delphi || \
+		docker build -t delphi_database -f repos/delphi/operations/dev/docker/database/Dockerfile .
+
+	@# Build the database_epidata image
+	@docker build -t delphi_database_epidata \
+		-f repos/delphi/delphi-epidata/dev/docker/database/epidata/Dockerfile .
+
+	@# Run the database
+	@docker run --rm -p 127.0.0.1:13306:3306 \
+		--network delphi-net --name delphi_database_epidata \
+		delphi_database_epidata >$(LOG_DB) 2>&1 &
+
+	@# Block until DB is ready
+	@while true; do \
+		sed -n '/Temporary server stopped/,/mysqld: ready for connections/p' $(LOG_DB) | grep "ready for connections" && break; \
+		tail -1 $(LOG_DB); \
+		sleep 1; \
+	done
+
+.PHONY=py
+py:
+	@# Build the python image
+	@docker build -t delphi_python \
+	-f repos/delphi/operations/dev/docker/python/Dockerfile .
+
+	@docker build -t delphi_web_python \
+	-f repos/delphi/delphi-epidata/dev/docker/python/Dockerfile .
+
+.PHONY=all
+all: web db py
+
+.PHONY=test
+test: 
+	@docker run -i --rm --network delphi-net \
+		--mount type=bind,source=$(CWD)repos/delphi/delphi-epidata,target=/usr/src/app/repos/delphi/delphi-epidata,readonly \
+		--mount type=bind,source=$(CWD)repos/delphi/delphi-epidata/src,target=/usr/src/app/delphi/epidata,readonly \
```

Comment:
Why is CWD needed here?

Reply:
This may have been one of those things that got me in the habit of using absolute paths 
```
docker: Error response from daemon: invalid mount config for type "bind": invalid mount path: 'repos/delphi/delphi-epidata' mount path must be absolute.
```